_posts/2021-02-11-tabnet/tabnet.Rmd

```diff
@@ -0,0 +1,411 @@
+
+# tidymodels
+
+To many R-using data scientists, the [tidymodels](https://www.tidymodels.org/) framework will not be a stranger. `tidymodels` provides a high-level, unified approach to model training, hyperparameter optimization, and inference.
+
+`tabnet` is the first (of many, we hope) `torch` models that let you use a `tidymodels` workflow all the way: from data pre-processing over hyperparameter tuning to performance evaluation and inference. While the first, as well as the last, may seem nice-to-have but not "mandatory", the tuning experience is likely to be something you'll won't want to do without anymore!
+
+# Using tabnet with tidymodels
+
+In this post, we first showcase a `tabnet`-using workflow in a nutshell, making use of hyperparameter settings reported in the paper.
+
+Then, we initiate a `tidymodels`-powered hyperparameter search, focusing on the basics but also, encouraging you to dig deeper at your leisure.
+
+Finally, we circle back to the promise of interpretability, demonstrating what is offered by `tabnet` and ending in a short discussion.
+
+# In the flow with TabNet
+
+As usual, we start by loading all required libraries. We also set a random seed, on the R as well as the `torch` sides. When model interpretation is part of your task, you will want to investigate the role of random initialization.
+
+```{r}
+library(torch)
+library(tabnet)
+library(tidyverse)
+library(tidymodels)
+library(finetune) # to use tuning functions from the new finetune package
+library(vip) # to plot feature importances
+
+set.seed(777)
+torch_manual_seed(777)
+
+```
+
+Next, we load the dataset.
+
+```{r}
+# download from https://archive.ics.uci.edu/ml/datasets/HIGGS
+higgs <- read_csv(
+  "HIGGS.csv",
+  col_names = c("class", "lepton_pT", "lepton_eta", "lepton_phi", "missing_energy_magnitude",
+                "missing_energy_phi", "jet_1_pt", "jet_1_eta", "jet_1_phi", "jet_1_b_tag",
+                "jet_2_pt", "jet_2_eta", "jet_2_phi", "jet_2_b_tag", "jet_3_pt", "jet_3_eta",
+                "jet_3_phi", "jet_3_b_tag", "jet_4_pt", "jet_4_eta", "jet_4_phi", "jet_4_b_tag",
+                "m_jj", "m_jjj", "m_lv", "m_jlv", "m_bb", "m_wbb", "m_wwbb"),
+  col_types = "fdddddddddddddddddddddddddddd"
+  )
+```
+
+What's this about? In high-energy physics, the search for new particles takes place at powerful particle accelerators, such as (and most prominently) CERN's [Large Hadron Collider](https://home.cern/science/accelerators/large-hadron-collider). In addition to actual experiments, simulation plays an important role. In simulations, "measurement" data are generated according to different underlying hypotheses, resulting in distributions that can be compared with each other. Given the likelihood of the simulated data, the goal then is to make inferences about the hypotheses.
```

Comment:
Great paragraph—really clear and concise despite the technical content!

Reply:
wow, thanks for saying that!!